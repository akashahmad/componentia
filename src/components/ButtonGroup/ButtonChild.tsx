import * as React from "react";

import cx from "classnames";

import { Button } from "../Button/Button";

import { SizeContext } from "./context";

export function ButtonGroupChild({ children, active = false, onClick, ...other }: ButtonChildProps) {
    const { value: size } = React.useContext(SizeContext);

    return (
        <div className={cx("border-r last:border-r-0")}>
            <Button
                size={size}
                onClick={onClick}
                variant={active ? "default" : "outlined"}
                className={cx("border-transparent rounded-none border-r")}
                style={!active ? { color: "#111928" } : { color: "white" }}
                {...other}
            >
                {children}
            </Button>
        </div>
    );
}

interface ButtonChildProps {
    active?: boolean;
    children: React.ReactNode;
    onClick?: () => void;
}
