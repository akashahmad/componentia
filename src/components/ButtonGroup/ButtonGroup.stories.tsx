import * as React from "react";

import type { Meta, StoryObj } from "@storybook/react";

import { Button } from "../Button/Button";
import { ButtonGroup } from "../ButtonGroup/ButtonGroup";

const meta: Meta<typeof Button> = {
    component: ButtonGroup,
    tags: ["autodocs"],
};

export default meta;
type GroupStory = StoryObj<typeof Button>;

export const Group: GroupStory = {
    args: {
        size: "Large",
    },
    argTypes: {},
    render: props => {
        const [active, setActive] = React.useState(0);

        return (
            <ButtonGroup {...props}>
                <ButtonGroup.Child onClick={() => setActive(0)} active={active === 0}>
                    About
                </ButtonGroup.Child>
                <ButtonGroup.Child onClick={() => setActive(1)} active={active === 1}>
                    Profile
                </ButtonGroup.Child>
                <ButtonGroup.Child onClick={() => setActive(2)} active={active === 2}>
                    Services
                </ButtonGroup.Child>
            </ButtonGroup>
        );
    },
};
