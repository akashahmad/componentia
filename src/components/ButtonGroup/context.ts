import * as React from "react";

import { sizePropType } from "./ButtonGroup";

interface SizeContextType {
    value: sizePropType;
}

export const SizeContext = React.createContext<SizeContextType>({ value: "Large" });
