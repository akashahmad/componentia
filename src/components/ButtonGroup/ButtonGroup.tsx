import * as React from "react";

import cx from "classnames";

import { ButtonGroupChild } from "./ButtonChild";

import { SizeContext } from "./context";

export function ButtonGroup({ children, size = "Large" }: ButtonGroupProps) {
    return (
        <SizeContext.Provider value={{ value: size }}>
            <div className={cx("inline-block border rounded-md overflow-hidden")}>
                <div className={cx("flex items-center")}>{children}</div>
            </div>
        </SizeContext.Provider>
    );
}

export type sizePropType = "Large" | "Small" | "Medium";

export interface ButtonGroupProps {
    children: React.ReactNode;
    size?: sizePropType;
}

ButtonGroup.Child = ButtonGroupChild;
