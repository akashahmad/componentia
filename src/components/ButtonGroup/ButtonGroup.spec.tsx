import * as React from "react";

import { render } from "@testing-library/react";
import { ButtonGroup } from "./ButtonGroup";

test("group-button component is rendering properly", () => {
    const { container } = render(
        <ButtonGroup>
            <ButtonGroup.Child>About</ButtonGroup.Child>
            <ButtonGroup.Child active>Services</ButtonGroup.Child>
        </ButtonGroup>,
    );

    expect(container).toMatchSnapshot();
});
