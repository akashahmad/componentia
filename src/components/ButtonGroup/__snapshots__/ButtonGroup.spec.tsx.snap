// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group-button component is rendering properly 1`] = `
<div>
  <div
    class="inline-block border rounded-md overflow-hidden"
  >
    <div
      class="flex items-center"
    >
      <div
        class="border-r last:border-r-0"
      >
        <button
          class="rounded-md px-7 py-3 border font-medium flex justify-center items-center text-base px-7 py-3 loading-6 text-primary border-primary hover:bg-blue-light-5 cursor-pointer border-transparent rounded-none border-r"
          role="button"
          style="color: rgb(17, 25, 40);"
        >
          About
        </button>
      </div>
      <div
        class="border-r last:border-r-0"
      >
        <button
          class="rounded-md px-7 py-3 border font-medium flex justify-center items-center text-base px-7 py-3 loading-6 bg-primary text-white hover:bg-blue-dark cursor-pointer border-transparent rounded-none border-r"
          role="button"
          style="color: white;"
        >
          Services
        </button>
      </div>
    </div>
  </div>
</div>
`;
